### PR TITLE
[5.7] Make Mockery a non-dev dependency

### PR DESCRIPTION
**Pull request by svenluijten**
`mockery/mockery` is being used in [`Illuminate\Foundation\Testing\PendingCommand`](https://github.com/laravel/framework/blob/5.7/src/Illuminate/Foundation/Testing/PendingCommand.php#L160), which can be used when not developing the internals of the framework in this repository.

This fixes issues I was having (`class Mockery not found`) when working on one of my packages that requires `laravel/framework`, but doesn't have Mockery anywhere in its own dependency graph. It should be in Laravel's. 🙂 

When creating a new Laravel application, `composer why mockery/mockery` gives the following output: 

```
λ composer why mockery/mockery
laravel/laravel  -  requires (for development)  mockery/mockery (^1.0)
```

This hasn't broken anyone's apps, but this will break when trying to use the `PendingCommand` outside of a regular application, like in a package 😄

### CODE REVIEW

**Comment by X-Coder264** [comment]
This class is there only to be used when testing Artisan commands, that's why it's in the Foundation **Testing** namespace.

While you are at it why didn't you move also PHPUnit from require-dev to require as PHPUnit is also required for that class? 😄 

TL;DR: Sorry, but I don't see how this change makes any sense.

**Comment by svenluijten** [comment]
@X-Coder264 When developing something with `laravel/framework` as a dependency (eg. a package), testing Artisan commands is impossible now. Laravel relies on the `Mockery` class to exist, so it should be a regular dependency of the framework.

The reason nobody has run into a problem before is because Mockery is so ubiquitous in the community (and it comes with a standard install of `laravel/laravel`), that that class just always existed. In this package I'm working on, Mockery doesn't exist, so I can't use the (new) [console tests](https://laravel.com/docs/5.7/console-tests) that were introduced in 5.7.

**Comment by svenluijten** [comment]
@X-Coder264 I see your point about also moving PHPUnit into the regular dependencies, and that got me thinking. What if (in case this PR gets rejected), we also put `phpunit/phpunit` and `mockery/mockery` in the `suggests` object in `composer.json`?

It would still feel a bit odd to have the consuming user (package, app, whatever...) "fill in" a dependency of the framework, but I could live with the `suggests` thing I suppose 🙂

**Comment by devcircus** [comment]
"composer require" whatever you need. We definitely shouldn't pull Mockery into non-dev environments by default.

**Comment by vlakoff** [comment]
Agree with the people above.

**Comment by svenluijten** [comment]
@devcircus see, I both agree and disagree. And I disagree because _I_ don't need Mockery, the framework does. So I'd be satisfying the dependencies of the framework.

What if Laravel stops using Mockery here? Then I'd have a stale dependency in my project because it's not being used anywhere. Just _feels_ weird, y'know? 

Any thoughts on this, should I just ignore the feeling and `composer require mockery/mockery --dev` anyway? 😕

**Comment by svenluijten** [comment]
Funny how you also ignored my question above, but whatever 🤷‍♂️ 

Thanks for the link in the documentation, I must've read over that. I suppose I'll have to find another way to get this working! Thanks for the input, all 🙂

**Comment by X-Coder264** [comment]
Sorry, but I don't see to what question are you referring. If you are talking about 

> What if Laravel stops using Mockery here?

That question is made obsolete just by reading my answer as to not require the entire `laravel/framework` package.

**Comment by svenluijten** [comment]
https://github.com/laravel/framework/pull/25994#issuecomment-427691572

**Comment by X-Coder264** [comment]
Oh, that. I don't see the point of adding them to `suggests` in composer. That would imply that you could use the `laravel/framework` package in the way that you wanted to use it in the first place which is not the way it's supposed to be used so we shouldn't encourage that. Both PHPUnit and Mockery are declared (correctly) as dev dependencies and that should remain that way.

**Comment by svenluijten** [comment]
Yeah makes sense. Thanks for taking your time to explain this! 👍

**Comment by X-Coder264** [comment]
Np, I'm glad that I could be of help (even if it's wasn't the kind of help that you probably wanted to hear) 😄
